api/v1alpha1/shared_types.go

```diff
@@ -121,7 +122,7 @@ func (in Expose) validate(path *field.Path) field.ErrorList {
 
 func (in *TransparentProxy) toConsul() *capi.TransparentProxyConfig {
 	if in == nil {
-		return &capi.TransparentProxyConfig{OutboundListenerPort: 0}
+		return nil
```

Comment:
Here is the change to write it to Consul as nil if we receive it as nil (which caused the other issues).